src/lib/Metadata/Reduce.cpp

```diff
@@ -298,6 +299,27 @@ void merge(FriendInfo& I, FriendInfo&& Other)
         I.FriendType = std::move(Other.FriendType);
 }
 
+void merge(AliasInfo& I, AliasInfo&& Other)
+{
+    MRDOCS_ASSERT(canMerge(I, Other));
+    if(! I.AliasedSymbol)
+        I.AliasedSymbol = std::move(Other.AliasedSymbol);
+    I.FullyQualifiedName = std::move(Other.FullyQualifiedName);
+    mergeSourceInfo(I, std::move(Other));
+    mergeInfo(I, std::move(Other));
+}
+
+void merge(UsingInfo& I, UsingInfo&& Other)
+{
+    MRDOCS_ASSERT(canMerge(I, Other));
+
+    reduceSymbolIDs(I.UsingSymbols, std::move(Other.UsingSymbols));
+    I.Class = Other.Class;
```

Comment:
You should only set `I.Class` to `Other.Class` if `I.Class != UsingClass::Normal`. Same with `I.Qualifier` (only set it to `Other.Qualifier` if `I.Qualifier != nullptr`)